optimade/validator/validator.py

```diff
@@ -286,7 +287,9 @@ def validate_implementation(self):
             entry_info_endpoint = f"{info_endp}/{endp}"
             self._log.debug("Testing expected info endpoint %s", entry_info_endpoint)
             entry_info = self._test_info_or_links_endpoint(entry_info_endpoint)
-            if entry_info:
+            if entry_info is None or isinstance(entry_info, dict):
+                self._entry_info_by_type[endp] = entry_info
```

Comment:
I don't know if this will be too much work, but it would be good to check that the validator is fine with this, specifically when `entry_info = None`. The other should be fine.

Reply:
Would be pretty difficult to inject this, but the only function that uses the entry info (`_recurse_through_endpoint`) now begins with an `if not entry_info: raise` so that the appropriate error can be raised safely outside of the top-level.

Reply:
I think it might actually be cleaner to just wrap this in a try/except for accessing `.dict()`, will push another commit

Reply:
So what kind of response would result in this situation then?
In any case, I think you're right about the try/except.

Reply:
This came up when OQMD's entry info was just missing the `meta` response, although all the other required `data` fields were there, so it could have been used to test the corresponding entry endpoint.

Reply:
Cool. So could you not feed such a response to a test and see that you get back what you expect from the validator?

Reply:
I've gone back on this approach, if the entry info cannot be deserialized then it will not try to proceed with the bare JSON. Thought we could get away with it but it turns out to be quite the rabbithole.

I agree it would be nice to add some more unit tests to the validator now that it has been modularised, but perhaps that is for another PR...